### ISSUE TITLE
Consider supporting Micronaut 4.0 in Helidon 4.x?

### ISSUE THREAD

**Issue by tjquinno** [helidon-io/helidon#7206]
## Environment Details
* Helidon Version: 4.x
* Helidon SE or Helidon MP
* JDK version:
* OS:
* Docker version (if applicable):

----------

## Problem Description
Micronaut has just released 4.0 (11 July 2023) which, among other things, has breaking changes regarding `javax` vs. `jakarta` in some areas including validation, as described in this excerpt from Slack:
> I'm having issue of imcompatibilty of the jakarta.validation and javax.validation.
The `jakarta.validation` (I'm refering to the constraints `import jakarta.validation.constraints.*` in to the model) doesn't work with micronaut `@Validated` in `CrudRepository` as it does not trigger the validation due the `helidon.integrations.micronaut` it only support` javax.validation` instead (tested it work).  
>  
> ```  
> version.lib.micronaut>3.4.3</version.lib.micronaut>, 
> version.lib.micronaut.data>3.3.0</version.lib.micronaut.data>  
> ``` 
> However with the use of `javax.validation.*` it create issue for manual validation, the `Validation.buildDefaultValidatorFactory() `doest work with` javax`, which trigger the error.  
>  
> ```  
> 1:57:56.883 [helidon-server-1] ERROR c.h.o.o.s.c.e.GlobalExceptionHandler - Unable to create a Configuration, because no Jakarta Bean Validation provider could be found. Add a provider like Hibernate Validator (RI) to your classpath.  
> javax.validation.NoProviderFoundException: Unable to create a Configuration, because no Jakarta Bean Validation provider could be found. Add a provider like Hibernate Validator (RI) to your classpath.
> ```

As Micronaut 4.0 contains breaking changes from Micronaut 3.x, we probably would not want to change the Micronaut dependency to 4.0 in a Helidon 3.x dot release.

But do we want to consider updating the Micronaut dependency to 4.0 in Helidon 4.x?

**Comment by ASoftwareProgramer** [comment]
These are JAVA JDK **java.compiler** library, would this impact the integration ? I believe these namespace will not be change for the JAVA platform.

**Comment by romain-grecourt** [comment]
> These are JAVA JDK **java.compiler** library, would this impact the integration ? I believe these namespace will not be change for the JAVA platform.

Correct, `jakarta.annotation:jakarta.annotation-api:jar:2.1.1` is not the replacement of `javax.annotation`.

**Comment by ASoftwareProgramer** [comment]
Hi All, 

I would like to know when will this fix to be release ?

**Comment by ASoftwareProgramer** [comment]
Hi All,

Any expected date to be release the Micronut integration ?

**Comment by jbescos** [comment]
> Hi All,
> 
> Any expected date to be release the Micronut integration fix ?

There is a [PR](https://github.com/helidon-io/helidon/pull/8230) but it is pending to be reviewed.
